### ISSUE TITLE
Design configuration (2.1.0) does not save

### ISSUE THREAD

**Issue by psiegel404** [magento/magento2#5666]
Hello,
thank you for the effort you are making.
i have issue in Content -> Desing -> Configuration in 2.1.0 version.

Steps to reproduce

Browse Content -> Design -> Configuration.
Click Edit Global.
Change any value and then Save Configuration.
Expected result

Save and refresh the page
Actual result

dose not save and show error: Cannot read contents from file "path/to/magento/" Warning!file_get_contents(path/to/magento/): failed to open stream: No such file or directory
web server info:

Magento 2.1.0
XAMPP 3.2.2 (Apache 2.4.18) 
Windows 7

and also

Magento 2.1.0 
XAMPP 3.2.2 (Apache 2.4.17)
Windows 10

thank you.

**Comment by JDavidVR** [comment]
Hi @kurtjmeyoung, you probably have permission problem, it works on linux.
And magento doesn't support officially windows, you can check here: http://devdocs.magento.com/guides/v2.0/install-gde/system-requirements.html

**Comment by developer-lindner** [comment]
@JDavidVR It's not a windows issue nor a permission problem.

**Comment by southerncomputer** [comment]
I've definitely checked this out using 2.1.0 under centos 7 /nginx/apache/php-fpm stack and it works quite fine, unless you are using a non magento2 theme that has issues!

**Comment by vivek201** [comment]
Yes, this is definitely an issue. The previous Magento 2.0.7 was working perfectly fine.
Has anyone found a solution to this?

**Comment by abdel1220** [comment]
Hello,
i had this problem and i already create ticket for that [#5288](https://github.com/magento/magento2/issues/5288)

and did't found solution yet.

**Comment by pboisvert** [comment]
closing as duplicate of #5288

**Comment by vivek201** [comment]
@pboisvert 
I still think that this is an issue and that it needs to be handled.
So far, I've found that this happens when the default email template is being loaded in the file  `magento2/app/code/Magento/Email/Model/AbstractTemplate.php`  in the function `loadDefault($templateId)` and the `$templateId` being passed is `design_email_header_template`.

In my understanding, the `$templateFile` variable in that function should hold the absolute path of the template file but instead it is coming out empty.

Also, saying that Windows is not supported is definitely not a solution.

Any help is appreciated, Thanks.

**Comment by vivek201** [comment]
Hello, 
I have found a workaround for this issue. Copy the folder `\vendor\magento\module-email\view\frontend\email` to `\vendor\magento\module-email\view\adminhtml\email\`.
After copying the folder, I can now save the configuration. Don't really know why. 
It would be really helpful if someone can explain why.
Thanks.

**Comment by devDomM** [comment]
wait? why was this closed when the "already closed duplicated" #5288 mentions "_ey please create a new issue!_" 

anyway we noticed something important:
- before you make changes to the design configuration you have to choose between "**Global**", the **Websites** you created, and also the **Store-Views**
- changes done in **Global** result with the explained error
- changes done in a created **Website** can be saved without an error

TL;DR: To prevent the error save your configuration on the Website-Level, not Global

and btw thanks @vivek201 for the workaround! will also try that.

**Comment by stamster** [comment]
You really should not use Windows for development. If you can't afford dedicated Linux box on your own, you can run it inside Windows via VirtualBox or similar non bare-metal hypervisor software. Save yourself some headache. 

Magento is a large project where there are plenty of issues already, so you don't need another layer to worry about such as 'video games' OS. 

Bottom line: officially Magento does not support M$ Windows, but only **Operating systems (Linux x86-64)**

**Comment by vivek201** [comment]
@devDomM Thanks, that actually works! Should've tried that before doing anything else.
@stamster Yeah, you're definitely right about the headaches. Just finished setting up a VirtualBox environment with Ubuntu. Hope, it works there.
Thanks guys.

**Comment by pboisvert** [comment]
reopening issue as #5288 was requested to have a new ticket opened and i closed this as a duplicate of that ticket. Hence this one should still be active.

**Comment by nanda1990** [comment]
i want to add wordpress blog in magento-2 sub folder but it is showing 404 not found error. I am using nginx web server and also define location bt still not showing. and my nginx confile is nginx.sample.conf file

**Comment by stamster** [comment]
You have't configured nginx then to route url to your wp phy path. 
Pay attention to `php` location blocks as well.

**Comment by Joe-Whitehead** [comment]
Still getting this issue in Magento 2.1.5. 
Solution provided by @vivek201 still works but can this fix be pulled in to a Magento release?

**Comment by orlangur** [comment]
@Joe-Whitehead, there is no reason to use Windows for development purposes thus it is not supported by Magento as already mentioned multiple times in this thread. Did you try some Docker-based or Ubuntu/CentOS in VirtualBox development environment on your Windows machine?

There is no reason to pull any fix until Windows compatibility is officially declared by Magento.

Most likely good pull requests from community members which introduce Windows compatibility not breaking other environment cases would be accepted. But such compatibility could be broken with any next change without CI enforcement.

**Comment by tamount** [comment]
HELLO
i want try the workaround . "Copy the folder \vendor\magento\module-email\view\frontend\email to \vendor\magento\module-email\view\adminhtml\email\." but i don ' thave this folders in my root directory, i have magento 2.1.5, php5.6 
In my folder " C:\xampp\htdocs\magento2\vendor\magento " i have: "Compose", "magento-composer-installer" and "zendframwork1", 
Can you help me what to do? pleaaaaaz

**Comment by IlnitskiyArtem** [comment]
@kurtjmeyoung, Thanks for the feedback. 
Unfortunately, I could not reproduce the issue in 2.1.7 , 2.1-develop, 2.2.0(develop)
May please let me know if this issue relevant for latest release of Magento 2?

**Comment by korostii** [comment]
Hi @IlnitskiyArtem, the [fix](https://github.com/magento/magento2/commit/bdf62301f6e5d58bc1d53de0f471de54ec04e9e4) seems to be present on branch `develop` only, so the issue should still be present on both 2.1.7 and `2.1-develop`

**Comment by ghost** [comment]
So now we're saying its Windows fault because Magento can't load a template yet works for everything else? This framework is becoming more concerning every day.

**Comment by korostii** [comment]
Well, they're saying they owe you *no* support for Windows since the Magento, Inc. didn't ever promise that Magento CE will be working on Windows out of the box (it just so happens that it does, treat as a pleasant coincidence).

It's a convenient excuse to ignore the bugs that cannot be reproduced on a Linux machine (which is, of course, a bit off-putting to say the least, considering that PHP itself is cross-platform). 

However, this issue doesn't happen to be one of those.

**Comment by orlangur** [comment]
@xAvarice working with _any_ framework it is a good thing to learn [system requirements](http://devdocs.magento.com/guides/v2.1/install-gde/system-requirements-tech.html) and follow them.

I would not say "Windows fault", this is more like inconsistency with Linux in this aspect. It would be fixed & enforced by core team if it was ever needed. Microsoft introduced Ubuntu bash shell support in Windows 10, probably, at some point Windows will have enough compatibility by itself without any code change from Magento side.

Not sure why there is still no check similar to PHP version which would forbid running Magento under unsupported Windows OS. It would really eliminate a lot of confusion and similar "issue" reports.

**Comment by stamster** [comment]
While officially there is no support for Windows systems (and it shouldn't matter as no one will run that in a production anyways) and honestly today we have loads of options for development - virtual box to start with. So anyone can happily still run it's Windows' but with only difference being running server for those products like Magento inside of VM.
Waiting for M$ to support something from Linux natively feels like a fairy tale to me, and it could take years to have it polished etc.

So on this I agree with @orlangur - simply add check for OS during app startup / installation and eventually at least those kinds of odd OS bugs will be avoided.

**Comment by likemusic** [comment]
Windows 7 + nginx 1.12 + php-fpm + Magento v2.2.2 - bug still exists.

Fortunately workaround still work too - https://github.com/magento/magento2/issues/5666#issuecomment-233195345

**Comment by davidbrandastic** [comment]
Affects 2.2.3 + Windows + Apache as well unless saving at a store view level.

**Comment by chandirasekarid** [comment]
@vivek201 

enter the informations in transaction emails section before saving the design configuration
It worked for me
![capture](https://user-images.githubusercontent.com/42635948/44511370-e2042480-a6d4-11e8-9b39-4c22b10e53fe.PNG)
